Failed to run e2e tests for ibmcloud with shouldProvisionCluster false

By default, TEST_E2E_PROVISION is false to run e2e tests. So shouldProvisionCluster is also false.

In such case, e2e tests failed with following error:
```
14:44:23  PodVM name:  simple-peer-pod
14:44:23  --- FAIL: TestCreateSimplePod (45.28s)
14:44:23      --- FAIL: TestCreateSimplePod/Simple_Peer_Pod (45.28s)
14:44:23          --- FAIL: TestCreateSimplePod/Simple_Peer_Pod/PodVM_is_created (0.00s)
14:44:23  panic: runtime error: invalid memory address or nil pointer dereference [recovered]
14:44:23  	panic: runtime error: invalid memory address or nil pointer dereference
14:44:23  [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1ffba46]
14:44:23  
14:44:23  goroutine 364 [running]:
14:44:23  testing.tRunner.func1.2({0x23c2a40, 0x36bf370})
14:44:23  	/usr/local/go/src/testing/testing.go:1396 +0x24e
14:44:23  testing.tRunner.func1()
14:44:23  	/usr/local/go/src/testing/testing.go:1399 +0x39f
14:44:23  panic({0x23c2a40, 0x36bf370})
14:44:23  	/usr/local/go/src/runtime/panic.go:884 +0x212
14:44:23  github.com/IBM/vpc-go-sdk/vpcv1.(*VpcV1).GetEnableGzipCompression(...)
14:44:23  	/home/jenkins/go/pkg/mod/github.com/!i!b!m/vpc-go-sdk@v1.0.1/vpcv1/vpc_v1.go:173
14:44:23  github.com/IBM/vpc-go-sdk/vpcv1.(*VpcV1).ListInstancesWithContext(0x0, {0x29c8050?, 0xc00004e0c0}, 0xc001330f00)
14:44:23  	/home/jenkins/go/pkg/mod/github.com/!i!b!m/vpc-go-sdk@v1.0.1/vpcv1/vpc_v1.go:4320 +0x146
14:44:23  github.com/IBM/vpc-go-sdk/vpcv1.(*VpcV1).ListInstances(...)
14:44:23  	/home/jenkins/go/pkg/mod/github.com/!i!b!m/vpc-go-sdk@v1.0.1/vpcv1/vpc_v1.go:4308
14:44:23  github.com/confidential-containers/cloud-api-adaptor/test/e2e.IBMCloudAssert.HasPodVM({0xc0000bcef0?}, 0xc0013501a0?, {0x2617c44?, 0xf?})
14:44:23  	/home/jenkins/agent/workspace/cloud-api-adaptor-e2e-tests/cloud-api-adaptor/test/e2e/ibmcloud_test.go:33 +0xd6
14:44:23  github.com/confidential-containers/cloud-api-adaptor/test/e2e.doTestCreateSimplePod.func2({0x29c8050, 0xc00004e0c0}, 0x51b?, 0x0?)
14:44:23  	/home/jenkins/agent/workspace/cloud-api-adaptor-e2e-tests/cloud-api-adaptor/test/e2e/common_suite_test.go:41 +0x3d
```

It looks like `pv.IBMCloudProps.VPC` is not initialized if shouldProvisionCluster is false, and `c.vpc.ListInstances(options)` failed.

--- a/test/e2e/main_test.go
+++ b/test/e2e/main_test.go
@@ -60,13 +60,11 @@ func TestMain(m *testing.M) {
 	//  key2 = "value2"
 	provisionPropsFile := os.Getenv("TEST_E2E_PROVISION_FILE")
 
-	if shouldProvisionCluster || podvmImage != "" {
-		// Get an provisioner instance for the cloud provider.
-		provisioner, err = pv.GetCloudProvisioner(cloudProvider, provisionPropsFile)
-		if err != nil {
-			fmt.Println(err)
-			os.Exit(1)
-		}
+	// Get an provisioner instance for the cloud provider.
+	provisioner, err = pv.GetCloudProvisioner(cloudProvider, provisionPropsFile)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
 	}
 
 	if !shouldProvisionCluster {

--- a/test/provisioner/provision_ibmcloud_initializer.go
+++ b/test/provisioner/provision_ibmcloud_initializer.go
@@ -149,7 +149,7 @@ func initProperties(properties map[string]string) error {
 	}
 
 	podvmImage := os.Getenv("TEST_E2E_PODVM_IMAGE")
-	if len(podvmImage) >= 0 {
+	if len(podvmImage) > 0 {
 		if len(IBMCloudProps.CosApiKey) <= 0 {
 			return errors.New("COS_APIKEY was not set.")
 		}

